autorest/azure/token.go

```diff
@@ -104,6 +154,79 @@ func (spt *ServicePrincipalToken) EnsureFresh() error {
 
 // Refresh obtains a fresh token for the Service Principal.
 func (spt *ServicePrincipalToken) Refresh() error {
+	newToken, err := spt.GetOauthToken()
+	if err != nil {
+		return err
+	}
+	spt.Token = *newToken
+
+	return nil
+}
+
+// SetAuthValues is a method of the interface ServicePrincipalTokenSecret
+// It will populate the form submitted during oAuth Token Acquisition using a static client_secret
+func (tokenSecret *ServicePrincipalTokenSecret) SetAuthValues(spt *ServicePrincipalToken, v *url.Values) error {
+	v.Set("client_secret", tokenSecret.ClientSecret)
+	return nil
+}
+
+// SignedJwt returns the JWT signed with the secret's certData
+func (secret *ServicePrincipalCertSecret) SignedJwt(spt *ServicePrincipalToken) (string, error) {
```

Comment:
Also, I prefer verbs for names. That is, I believe this is better called "SignJwt" vs. "SignedJwt". SignedJwt is a noun that defines a thing. SignJwt is an action that describes the method.